src/cmd/ksh93/sh/io.c

```diff
@@ -532,21 +532,26 @@ static int inetopen(const char *path, int flags, Inetintr_f onintr, void *handle
             return -1;
         }
     }
+
     if (flags == O_NONBLOCK) return 1;
-    if (!(s = strdup(path))) return -1;
-    t = strchr(s, '/');
-    if (t) {
-        *t++ = 0;
-        if (!strcmp(s, "local")) s = strdup("localhost");
-        fd = getaddrinfo(s, t, &hint, &addr);
-    } else {
-        fd = -1;
+
+    t = strchr(path, '/');
+    if (!t) return -1;
+
+    s = strdup(path);
```

Comment:
Shall we check if `strdup()` fails ?

Reply:
Not really much point in an explicit `assert(s)`. If the following assignment doesn't fail (say because we're running on a system which doesn't make page zero invalid) the `strcmp()` will definitely fail.